Sources/Commands/SwiftTestTool.swift

```diff
@@ -211,11 +237,70 @@ public struct SwiftTestTool: SwiftTool {
         return (mode ?? .run(nil), opts)
     }
 
-    private func test(path: AbsolutePath, xctestArg: String? = nil) throws -> Bool {
-        guard isValidTestPath(path) else {
-            throw TestError.testsExecutableNotFound
+    /// Executes the tests in parallel.
+    ///
+    /// - Parameters:
+    ///     - testPath: Path to a valid XCTest binary.
+    ///
+    /// - Returns: True if all the tests exited with return code zero.
+    private func testInParallel(testPath: AbsolutePath) throws -> Bool {
+        precondition(isValidTestPath(testPath))
+
+        // Create a queue to hold specifiers.
+        let queue = SynchronizedQueue<String?>()
+
+        // Find all the test suites present in the test binary.
+        let testSuites = try getTestSuites(path: testPath)
+
+        // Enqueue all the tests.
+        for testSuite in testSuites {
```

Comment:
testSuites.flatMap {$0.tests}.flatMap {{$0.tests} - would it work ?
But maybe it would an overhead for compiler to build such complex expression